azurerm/resource_arm_dns_ptr_record.go

```diff
@@ -0,0 +1,169 @@
+package azurerm
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/Azure/azure-sdk-for-go/arm/dns"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceArmDnsPtrRecord() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmDnsPtrRecordCreate,
+		Read:   resourceArmDnsPtrRecordRead,
+		Update: resourceArmDnsPtrRecordCreate,
+		Delete: resourceArmDnsPtrRecordDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"resource_group_name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"zone_name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"records": &schema.Schema{
+				Type:     schema.TypeSet,
+				Required: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Set:      schema.HashString,
+			},
+
+			"ttl": &schema.Schema{
+				Type:     schema.TypeInt,
+				Required: true,
+			},
+
+			"tags": tagsSchema(),
+		},
+	}
+}
+
+func resourceArmDnsPtrRecordCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient)
+	dnsClient := client.dnsClient
+
+	name := d.Get("name").(string)
+	resGroup := d.Get("resource_group_name").(string)
+	zoneName := d.Get("zone_name").(string)
+	ttl := int64(d.Get("ttl").(int))
+	tags := d.Get("tags").(map[string]interface{})
+
+	metadata := expandTags(tags)
+
+	recordStrings := d.Get("records").(*schema.Set).List()
+	records := make([]dns.PtrRecord, len(recordStrings))
+
+	for i, v := range recordStrings {
+		fqdn := v.(string)
+		records[i] = dns.PtrRecord{
+			Ptrdname: &fqdn,
+		}
+	}
+
+	props := dns.RecordSetProperties{
+		Metadata:   metadata,
+		TTL:        &ttl,
+		PtrRecords: &records,
+	}
+
+	parameters := dns.RecordSet{
+		Name:                &name,
+		RecordSetProperties: &props,
+	}
+
+	_, err := dnsClient.CreateOrUpdate(resGroup, zoneName, name, dns.PTR, parameters, "", "")
+	if err != nil {
+		return err
+	}
+
+	rec, err := dnsClient.Get(resGroup, zoneName, name, dns.PTR)
```

Comment:
I believe the dnsClient.CreateOrUpdate method's "response" return value already contains the resource ID, there is no need to call Get just to retrieve the ID value.

```
rec, err := dnsClient.CreateOrUpdate(...)

if err != nil {
    return err
}
d.SetId(*rec.ID)
```